server/build.gradle

```diff
@@ -125,13 +125,13 @@ dependencies {
   api 'org.hdrhistogram:HdrHistogram:2.1.12'
 
   // lucene spatial
-  api "org.locationtech.spatial4j:spatial4j:${versions.spatial4j}", optional
-  api "org.locationtech.jts:jts-core:${versions.jts}", optional
```

Comment:
@nknize @reta build was failing due to `optional` flag out here for the dependency upgrade. Looking forward to understand the significance of removing `optional` here.

Reply:
@owaiskazi19 It seems like plugin that provided `optional` is gone in 10.0.0 (it was one of the Nebula's one), we need to carefully understand if replacement is need or it could be safely removed. The `optional` does impact transitive dependencies, fe removing it would automatically bring `org.locationtech.*` dependencies to everyone, even if those are not used.

Reply:
Makes sense. @DarshitChanpura has raised https://github.com/opensearch-project/OpenSearch/pull/7152 to address the above.

Reply:
@DarshitChanpura @owaiskazi19 here is the findings, removing `optional` clearly changes the Maven/Ivy style dependencies, here is what we have:

```
    <dependency>
      <groupId>org.locationtech.spatial4j</groupId>
      <artifactId>spatial4j</artifactId>
      <version>0.7</version>
      <scope>compile</scope>
      <exclusions>
        <exclusion>
          <artifactId>*</artifactId>
          <groupId>*</groupId>
        </exclusion>
      </exclusions>
      <optional>true</optional>
    </dependency>
```

vs what we would have with `optional` gone (notice `optional` is not present)

```
    <dependency>
      <groupId>org.locationtech.spatial4j</groupId>
      <artifactId>spatial4j</artifactId>
      <version>0.7</version>
      <scope>compile</scope>
      <exclusions>
        <exclusion>
          <artifactId>*</artifactId>
          <groupId>*</groupId>
        </exclusion>
      </exclusions>
    </dependency>

```

To fix that, reintroduced this missing functionality as `OptionalDependenciesPlugin` (many projects sadly found themselves in situation to follow this path, [1] fe) but the semantics of `optional` will be preserved

[1] https://github.com/spring-projects/spring-boot/blob/ce85290d01a4d425936c173cc63d565aeef8aabd/buildSrc/src/main/java/org/springframework/boot/build/optional/OptionalDependenciesPlugin.java#L17

Reply:
Missed this @reta. Thanks for digging in. I see the PR is merged already!